src/language/JSUtils.js

```diff
@@ -68,19 +57,99 @@ define(function (require, exports, module) {
     function _findAllFunctionsInText(text) {
         var results = {},
             functionName,
+            resultNode,
+            memberPrefix,
             match;
-
+   
         PerfUtils.markStart(PerfUtils.JSUTILS_REGEXP);
-
-        while ((match = _functionRegExp.exec(text)) !== null) {
-            functionName = (match[2] || match[5]).trim();
-
+        
+        var AST = AcornLoose.parse_dammit(text, {locations: true});
+        
+        function _addResult(node, offset, prefix) {
+            memberPrefix = prefix ? prefix + " - " : "";
+            resultNode = node.id || node.key || node;
+            functionName = resultNode.name;
             if (!Array.isArray(results[functionName])) {
                 results[functionName] = [];
             }
 
-            results[functionName].push({offsetStart: match.index});
+            results[functionName].push(
+                {
+                    offsetStart: offset || node.start,
+                    label: memberPrefix ? memberPrefix + functionName : null,
+                    location: resultNode.loc
+                }
+            );
         }
+        
+        ASTWalker.simple(AST, {
+            /*
+                function <functionName> () {}
+            */
+            FunctionDeclaration: function (node) {
+                if (node.id.name !== '✖') { // A hack to reject anonymous declarations 
```

Comment:
Consider this snippet - 
``` javascript 
key: function () {
}
```
This is actually a broken syntax and acorn can't parse it, but acorn_loose does it. FYI the acorn_loose code which uses this as is - 
``` javascript 
LooseParser.prototype.dummyIdent = function dummyIdent () {
  var dummy = this.dummyNode("Identifier");
  dummy.name = "✖";
  return dummy
};
```
The problem here is `FunctionDeclaration` walk option catches this and if I don't have a check, ends up putting "✖" as a function name. To actually put `key` as function name we have a `LabeledStatement` walk option. For your reference this is how the node looks like - 
![image](https://user-images.githubusercontent.com/12087205/29662947-a7db07ae-88e6-11e7-984a-7b89e6e06a68.png)


This is not exactly a normal case but rather a broken syntax handler where this gets initialized with this name. I am not sure of any better way to handle this.